controllers/announcements.go

```diff
@@ -0,0 +1,102 @@
+package controllers
+
+import (
+    "strconv"
+    "net/http"
+    "io/ioutil"
+    "encoding/json"
+    "gopkg.in/macaron.v1"
+)
+
+type Topic struct {
+    ID int `json:"id"`
+    Title string `json:"title"`
+    Slug string `json:"slug"`
+}
+
+type TopicList struct {
+    Topics []Topic `json:"topics"`
+}
+
+type Category struct {
+    TopicList TopicList `json:"topic_list"`
+}
+
+type Post struct {
+    Cooked string `json:"cooked"`
+}
+
+type PostStream struct {
+    Posts []Post `json:"posts"`
+}
+
+type TopicResponse struct {
+    PostStream PostStream `json:"post_stream"`
+}
+
+type AnnouncementView struct {
+    First Announcement `json:"first"`
+    Second Announcement `json:"second"`
+}
+
+type Announcement struct {
+    Title string `json:"title"`
+    Content string `json:"content"`
+    Link string `json:"link"`
+}
+
+func GetAnnouncements(ctx *macaron.Context) {
+    r, err := http.Get("https://forums.spongepowered.org/c/announcements.json?order=created")
+    body, err := ioutil.ReadAll(r.Body)
+
+    if err != nil {
+        ctx.Error(http.StatusInternalServerError, "Can't access announcements!")
+        return
+    }
+
+    var res Category
+    json.Unmarshal(body, &res)
+
+    first, err := getAnnouncement(
+        res.TopicList.Topics[0].Title,
+        res.TopicList.Topics[0].ID,
+        res.TopicList.Topics[0].Slug)
+
+    if err != nil {
+        ctx.Error(http.StatusInternalServerError, "Can't access the first topic!")
+        return
+    }
+
+    second, err := getAnnouncement(
+        res.TopicList.Topics[1].Title,
+        res.TopicList.Topics[1].ID,
+        res.TopicList.Topics[1].Slug)
+
+    if err != nil {
+        ctx.Error(http.StatusInternalServerError, "Can't access the second topic!")
+        return
+    }
+
+    ctx.JSON(http.StatusOK, &AnnouncementView{
+        First: first,
+        Second: second,
+    })
+}
+
+func getAnnouncement(title string, id int, slug string) (Announcement, error) {
+    r, err := http.Get("https://forums.spongepowered.org/t/" + strconv.Itoa(id) + ".json")
+    body, err := ioutil.ReadAll(r.Body)
+
+    if err != nil {
+        return *&Announcement{}, err
```

Comment:
Lol :P You create an `Announcement`, take a pointer of it, and dereference it directly again, you can just do `Annoucement{}` directly then.